src/datasets/formatting/__init__.py

```diff
@@ -0,0 +1,128 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# flake8: noqa
+# Lint as: python3
+
+from typing import List, Optional, Union
+
+from ..utils.file_utils import is_tf_available, is_torch_available
+from ..utils.logging import get_logger
+from .formatting import (
+    CustomFormatter,
+    Formatter,
+    NumpyFormatter,
+    PandasFormatter,
+    PythonFormatter,
+    format_table,
+    query_table,
+)
+
+
+logger = get_logger(__name__)
+
+_FORMAT_TYPES = {}
+_FORMAT_TYPES_ALIASES = {}
+_FORMAT_TYPES_ALIASES_UNAVAILABLE = {}
+
+
+def register_formatter(format_type: Union[None, str], aliases: Optional[List[str]] = None):
```

Comment:
I started using it at a decorator at first but I switched to using it as a regular function.
I'll simplify it to be a regular function instead of a decorator, and also rename it _register_formatter